lib/sort-lines.js

```diff
@@ -14,12 +14,20 @@ function sortLines(textEditor, startLine, endLine, algorithm) {
   for (var i = startLine; i <= endLine; i++) {
     lines.push(textEditor.document.lineAt(i).text);
   }
-  lines.sort(algorithm);
-  textEditor.edit(function (editBuilder) {
-    for (var i = startLine; i <= endLine; i++) {
-      var line = lines[i - startLine];
-      editBuilder.replace(textEditor.document.lineAt(i).range, line);
+  var doUnique = (typeof(algorithm) == "boolean" && algorithm);
+  lines.sort(doUnique ? undefined : algorithm);
+  if (doUnique) {
+    var linesUnique = [];
+    for (var i=0; i<lines.length; ++i) {
+      if (linesUnique.length == 0 ||
+         (linesUnique[linesUnique.length-1] !== lines[i])) {
+        linesUnique.push(lines[i]);
+      }
     }
+    lines = linesUnique;
+  }
+  textEditor.edit(function (editBuilder) {
+    editBuilder.replace(textEditor.selection, lines.join("\n"));
```

Comment:
There is one thing this changes; it sorts the selection only, not the lines. So it can change the text if the selection does not cover the entire line. I'll fix it up.